lib/lint.js

```diff
@@ -0,0 +1,70 @@
+const { validateHubl } = require('@hubspot/cli-lib/api/validate');
+const { getPortalId } = require('@hubspot/cli-lib');
+const vscode = require('vscode');
+
+const getRange = (document, error) => {
+  if (error.startPosition > 0) {
+    return document.getWordRangeAtPosition(
+      new vscode.Position(error.lineno - 1, error.startPosition - 1)
+    );
+  } else {
+    return document.lineAt(new vscode.Position(error.lineno - 1, 0)).range;
+  }
+};
+
+async function updateDiagnostics(document, collection) {
+  const { renderingErrors } = await validateHubl(
+    getPortalId(),
+    document.getText()
+  );
+
+  const fitleredErrors = renderingErrors.filter((error) => {
+    return error.reason != 'MISSING';
+  });
+
+  const templateErrors = fitleredErrors.map((error) => {
+    return {
+      code: '',
+      message: error.message,
+      range: getRange(document, error),
+      severity: vscode.DiagnosticSeverity.Error,
+    };
+  });
+
+  if (document) {
+    collection.set(document.uri, templateErrors);
+  } else {
+    collection.clear();
+  }
+}
+
+const collection = vscode.languages.createDiagnosticCollection('hubl');
+let documentChangeListener;
+
+const enableLinting = () => {
+  let timeout = null;
+
+  if (vscode.window.activeTextEditor) {
+    updateDiagnostics(vscode.window.activeTextEditor.document, collection);
+  }
+
+  documentChangeListener = vscode.workspace.onDidChangeTextDocument(
+    (editor) => {
+      if (editor) {
+        clearTimeout(timeout);
+        timeout = setTimeout(function () {
+          updateDiagnostics(editor.document, collection);
+        }, 1000);
```

Comment:
I played around with it and the 1 second debounce feels pretty good to me, but I don't really have a good idea of what the standard is for linting. Have you looked into it? We should make sure we don't accidentally put out something that will cause vs code to work too hard.

Reply:
I took a sample of some other extension that lint code and it's all over the place (250-1000ms) but most plugins don't denounce at all because the linter process is running on the local machine instead of needing to make an API request.